articles/building-a-basic-api-with-nodejs/index.md

```diff
@@ -0,0 +1,220 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/building-a-basic-api-with-nodejs/
+title: How To Build and Test a Node.js REST API with Express on Ubuntu 18.04
+description: In this guide, you will build a Node.js REST API with the Express Framework, expose it to the internet with Ngrok.
+author: adrian-murage
+date: 2020-08-25T00:00:00-07:00
+topics: [Node.js]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/building-a-basic-api-with-nodejs/hero.jpg
+    alt: basic API with Node.js image example
+---
+In this guide, you will build a [Node.js](https://nodejs.dev/learn) REST API with the [Express Framework](http://expressjs.com/), expose it to the internet with [Ngrok](https://ngrok.com/docs) and make test requests to it on [Postman](https://www.postman.com/).
+<!--more-->
+
+### Introduction
+Web services based on REST Architecture are known as RESTful web services. These web services use HTTP methods to implement the concept of REST architecture. A RESTful web service usually defines a [URI](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier) which provides resource representation such as JSON and set of HTTP Methods.
+
+### Prerequisites
+Before you begin this guide you'll need the following:
+
+- A non-root user with sudo privileges on Ubuntu 18.04
```

Comment:
That's a plus but I intend it to be an Ubuntu 18.04 guide that's clear enough to teach the concepts. So that one who wishes to develop on a different platform gets the core concepts then uses documentation to implement it on their specific platform.